filebeat/scripts/tester/main.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func getLogsFromFile(logfile, multiPattern string, multiNegate bool) ([]string, error) {
+	regex, err := regexp.Compile(multiPattern)
+	if err != nil {
+		return nil, err
+	}
+
+	f, err := os.Open(logfile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var logs []string
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		l := getCompleteLine(s, multiNegate, regex)
+		logs = append(logs, l...)
+	}
+	return logs, nil
+}
+
+func getCompleteLine(s *bufio.Scanner, multiNegate bool, regex *regexp.Regexp) []string {
+	line := s.Text()
+	if regex.String() == "" {
+		return []string{line}
+	}
+	return getMultiline(s, line, multiNegate, regex)
+}
+
+func getMultiline(s *bufio.Scanner, line string, multiNegate bool, regex *regexp.Regexp) []string {
+	matches := regex.MatchString(line)
+	fullLine := line
+	if matches || !matches && multiNegate {
```

Comment:
Could we reuse here the logic of our multiline reader to make sure we have the exact same behaviour?

Reply:
Done.